piptools/scripts/compile.py

```diff
@@ -445,8 +453,17 @@ def cli(
             allow_unsafe=allow_unsafe,
         )
         results = resolver.resolve(max_rounds=max_rounds)
+
+        results_wheels = set()
+        hashes_wheels = dict()
+
+        if split_wheels:
+            results_wheels = set([result for result in results
+                                  if result.is_wheel])
+        results -= results_wheels
         if generate_hashes:
             hashes = resolver.resolve_hashes(results)
+            hashes_wheels = resolver.resolve_hashes(results_wheels)
```

Comment:
Is there any chance that doing this separately yields different results for the hashes? i.e. are the hashes dependent on the set of files provided to the resolver?

Reply:
yeah it seems the hashes are dependent on the set of files provided, it is keyed by the `InstalledRequirement`